### PR TITLE
[RFR] Prettier

### PR DESCRIPTION
**Pull request by djhi**
- [x] Upgrade Prettier
- [x] Ensure Prettier Config is complete to avoid user overrides and applied to all files
- [x] Apply prettier on the project

### CODE REVIEW

**Comment by fzaninotto** [comment]
We should not upgrade prettier before we are about to release v3. It would make the backport of the `master` changes to `next` extremely painful.

**Comment by djhi** [comment]
> We should not upgrade prettier before we are about to release v3. It would make the backport of the master changes to next extremely painful.

As discussed I will make the same PR on `next` to avoid any issues

**Comment by fzaninotto** [comment]
Can you revert the line width change to minimize the patch?

File: packages/ra-ui-materialui/src/list/List.js [hunk 1]
```diff
@@ -98,23 +98,9 @@ const sanitizeRestProps = ({
...
-    // overridable by user
-    children,
-    className,
-    classes,
-    exporter,
-    title,
-    ...rest
-}) => {
+export const ListView = withStyles(styles)(({ // component props
+    actions, aside, filter, filters, bulkActions, bulkActionButtons, pagination, children, className, classes, exporter, title, ...rest }) => {
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/List.js]: weird

Comment (djhi) [reply packages/ra-ui-materialui/src/list/List.js]: Looks like the comments are doing this

**Comment by fzaninotto** [comment]
Thanks! Next step, the `next` branch.

**Pull request merged**